symengine/visitor.cpp

```diff
@@ -175,4 +177,185 @@ void TransformVisitor::bvisit(const MultiArgFunction &x)
     auto nbarg = x.create(newargs);
     result_ = nbarg;
 }
+
+void preorder_traversal_local_stop(const Basic &b, LocalStopVisitor &v)
+{
+    b.accept(v);
+    if (v.stop_ or v.local_stop_)
+        return;
+    for (const auto &p : b.get_args()) {
+        preorder_traversal_local_stop(*p, v);
+        if (v.stop_)
+            return;
+    }
+}
+
+class IsALinearArgTrigVisitor
```

Comment:
Can you move this to `solve.cpp`

Reply:
cool. 
`IsALinearArgTrigVisitor` might not be used in any other situation, Is this the reason for moving?

Reply:
Yes, seems like it's needed for `solvers.cpp` only